source/omit-index-signature.d.ts

```diff
@@ -91,14 +93,23 @@ interface Example {
 	qux?: 'baz';
 }
 
-type ExampleWithoutIndexSignatures = RemoveIndexSignature<Example>;
+type ExampleWithoutIndexSignatures = OmitIndexSignature<Example>;
 // => { foo: 'bar'; qux?: 'baz' | undefined; }
 ```
 
+@see PickIndexSignature
 @category Object
 */
-export type RemoveIndexSignature<ObjectType> = {
+export type OmitIndexSignature<ObjectType> = {
 	[KeyType in keyof ObjectType as {} extends Record<KeyType, unknown>
 		? never
 		: KeyType]: ObjectType[KeyType];
 };
+
+export type {
+	/**
+	Alias to support versions prior to 3.
+	@deprecated Use {@link OmitIndexSignature} instead.
+	*/
+	OmitIndexSignature as RemoveIndexSignature,
+};
```

Comment:
I think it's better to put the alias in `index.d.ts` so we have them all in one place.